dev/sbt-plugin/src/sbt-test/sbt-plugin/akka-persistence-typed-migration-scala/shopping-cart-akka-persistence-typed/src/main/scala/com/example/shoppingcart/impl/ShoppingCartEntity.scala

```diff
@@ -0,0 +1,284 @@
+    cmd match {
+      case x: UpdateItem => onUpdateItem(x)
+      case x: Checkout   => onCheckout(x)
+      case x: Get        => onReadState(x)
+    }
+
+  def onUpdateItem(cmd: UpdateItem): ReplyEffect[ShoppingCartEvent, ShoppingCartState] =
+    cmd match {
+      case UpdateItem(_, qty, _) if qty < 0 =>
+        Effect.reply(cmd)(Rejected("Quantity must be greater than zero"))
+
+      case UpdateItem(productId, 0, _) if !items.contains(productId) =>
+        Effect.reply(cmd)(Rejected("Cannot delete item that is not already in cart"))
+
+      case UpdateItem(productId, quantity, _) =>
+        Effect
+          .persist(ItemUpdated(productId, quantity, Instant.now()))
+          .thenReply(cmd) { _ =>
+            Accepted
+          }
+    }
+
+
+  def onCheckout(cmd: Checkout): ReplyEffect[ShoppingCartEvent, ShoppingCartState] =
+    if (items.isEmpty)
+      Effect.reply(cmd)(Rejected("Cannot checkout empty cart"))
+    else
+      Effect
+        .persist(CheckedOut(Instant.now()))
+        .thenReply(cmd) { _ =>
+          Accepted
+        }
+
+  def onReadState(cmd: Get): ReplyEffect[ShoppingCartEvent, ShoppingCartState] =
+    Effect.reply(cmd)(CurrentState(this))
+
+  def applyEvent(evt: ShoppingCartEvent): ShoppingCartState = {
+    evt match {
+      case ItemUpdated(productId, quantity, _) => updateItem(productId, quantity)
+      case CheckedOut(_)                       => checkout
+    }
+  }
+
+  private def updateItem(productId: String, quantity: Int) = {
+    quantity match {
+      case 0 => copy(items = items - productId)
+      case _ => copy(items = items + (productId -> quantity))
+    }
+  }
+
+  private def checkout = copy(checkedOut = true)
+}
+
+object ShoppingCartState {
+
+  def empty: ShoppingCartState = ShoppingCartState(Map.empty, checkedOut = false)
+
+  val typeKey = EntityTypeKey[ShoppingCartCommand[_]]("ShoppingCartEntity")
+
+  def behavior(entityContext: EntityContext): Behavior[ShoppingCartCommand[_]] = {
+
+    val persistenceId = typeKey.persistenceIdFrom(entityContext.entityId)
+
+    EventSourcedBehavior
+      .withEnforcedReplies[ShoppingCartCommand[_], ShoppingCartEvent, ShoppingCartState](
+        persistenceId = persistenceId,
+        emptyState = ShoppingCartState.empty,
+        commandHandler = (cart, cmd) => cart.applyCommand(cmd),
+        eventHandler = (cart, evt) => cart.applyEvent(evt)
+      )
+      .withTagger(AkkaTaggerAdapter.fromLagom(entityContext, ShoppingCartEvent.Tag))
+
+  }
+
+  /**
+   * Format for the hello state.
+   *
+   * Persisted entities get snapshotted every configured number of events. This
+   * means the state gets stored to the database, so that when the entity gets
+   * loaded, you don't need to replay all the events, just the ones since the
+   * snapshot. Hence, a JSON format needs to be declared so that it can be
+   * serialized and deserialized when storing to and from the database.
+   */
```

Comment:
This implementation (the Akka Persistence entity) is a lot less verbose in terms of code comments but I still found some (like the one in lines 110-118) that doesn't seem to be specific to the `scripted` test. I tend to prefer keeping comments to a minimum set of comments for maintainers of the test.